### PR TITLE
fix/Tracking hook, context

### PR DESCRIPTION
**Pull request by marian-buc**
## What?
Removed tracking context and changed hook to call the callback directly after data is loaded

## Why?
To avoid querying the data where the hook is not used or having the queue go stale after rendering

**Ticket(s): []**

### CODE REVIEW

File: src/client/pages/Checkout/CheckoutDetails/CheckoutDetails.tsx [hunk 1]
```diff
@@ -26,14 +26,14 @@ export const CheckoutDetails = () => {
...
-  const { trackOfferEvent } = useTrackingContext()
+  const trackOfferEvent = useTrackOfferEvent()
   const data = useQuoteCartData()
 
-  useEffect(() => trackOfferEvent({ eventName: EventName.CheckoutOpen }), [
-    trackOfferEvent,
-  ])
+  const { quoteCartId } = useQuoteCartIdFromUrl()
+
+  useEffect(() => trackOfferEvent({ eventName: EventName.CheckoutOpen }), [])
```
Comment (robinandeer) [inline src/client/pages/Checkout/CheckoutDetails/CheckoutDetails.tsx:35]: I think we should add `trackOfferEvent` to the dependency array

File: src/routes.tsx [hunk 2]
```diff
@@ -425,5 +425,3 @@ export const routes: Route[] = [
     },
   },
 ]
-
-export const routePaths = routes.map((route) => route.path.toString())
```
Comment (robinandeer) [inline src/routes.tsx:429]: is this not imported anywhere?

**Review by robinandeer — COMMENTED** [review]
I'm not really sure what the motivation is here.

Didn't we already find some race conditions where we never send the analytics events?

File: src/routes.tsx (hunk 2 above)
Comment (dendoesit) [reply src/routes.tsx:429]: this was added so we can use it in useQuoteCartIdFromUrl but we are rolling all that logic back.

**Review by robinandeer — APPROVED** [review]
Looks good, now I understand!

**Pull request merged**